compose/snippets/src/main/java/com/example/compose/snippets/navigation3/scenes/ScenesSnippets.kt

```diff
@@ -73,132 +76,134 @@ public class SinglePaneSceneStrategy<T : Any> : SceneStrategy<T> {
-            val sceneKey = Pair(firstEntry.contentKey, secondEntry.contentKey)
-
-            TwoPaneScene(
-                key = sceneKey,
-                // Where we go back to is a UX decision. In this case, we only remove the top
-                // entry from the back stack, despite displaying two entries in this scene.
-                // This is because in this app we only ever add one entry to the
-                // back stack at a time. It would therefore be confusing to the user to add one
-                // when navigating forward, but remove two when navigating back.
-                previousEntries = entries.dropLast(1),
-                firstEntry = firstEntry,
-                secondEntry = secondEntry
-            )
-        } else {
-            null
-        }
+        val detailEntry =
+            entries.lastOrNull()?.takeIf { it.metadata.containsKey(DETAIL_KEY) } ?: return null
+        val listEntry = entries.findLast { it.metadata.containsKey(LIST_KEY) } ?: return null
+
+        // We use the list's contentKey to uniquely identify the scene.
+        // This allows the detail panes to be displayed instantly through recomposition, rather than
+        // having NavDisplay animate the whole scene out when the selected detail item changes.
+        val sceneKey = listEntry.contentKey
+
+        return ListDetailScene(
+            key = sceneKey,
+            previousEntries = entries.dropLast(1),
+            listEntry = listEntry,
+            detailEntry = detailEntry
+        )
     }
 }
 // [END android_compose_navigation3_scenes_3]
 
 // [START android_compose_navigation3_scenes_4]
 // Define your navigation keys
 @Serializable
-data object ProductList : NavKey
+data object ConversationList : NavKey
+
 @Serializable
-data class ProductDetail(val id: String) : NavKey
+data class ConversationDetail(
+    val id: Int,
+    val colorId: Int
+) : NavKey
 
 @Composable
 fun MyAppContent() {
-    val backStack = rememberNavBackStack(ProductList)
+    val backStack = rememberNavBackStack(ConversationList)
+    val listDetailStrategy = rememberListDetailSceneStrategy<NavKey>()
 
     NavDisplay(
         backStack = backStack,
+        onBack = { backStack.removeLastOrNull() },
+        sceneStrategy = listDetailStrategy,
         entryProvider = entryProvider {
-            entry<ProductList>(
-                // Mark this entry as eligible for two-pane display
-                metadata = TwoPaneScene.twoPane()
-            ) { key ->
-                Column {
-                    Text("Product List")
-                    Button(onClick = { backStack.add(ProductDetail("ABC")) }) {
-                        Text("View Details for ABC (Two-Pane Eligible)")
-                    }
-                }
+            entry<ConversationList>(
+                metadata = ListDetailScene.listPane()
+            ) {
+                // Add new Detail with backStack.addDetail(detailRoute)
+                Text(
```

Comment:
Actually, can you change this back to the old code i.e. a column with text explaining what the screen is showing and a button which adds the detail key to the back stack.